x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/three_way_diff/comparison_side/comparison_side.stories.tsx

```diff
@@ -0,0 +1,315 @@
+}
+
+const Template: Story<TemplateProps<keyof DiffableAllFields>> = (args) => {
+  return (
+    <ComparisonSide
+      fieldName={args.fieldName}
+      fieldThreeWayDiff={args.fieldThreeWayDiff}
+      resolvedValue={args.resolvedValue}
+    />
+  );
+};
+
+export const NoBaseVersion = Template.bind({});
+NoBaseVersion.args = {
+  fieldName: 'rule_name_override',
+  fieldThreeWayDiff: {
+    has_base_version: false,
+    base_version: undefined,
+    current_version: {
+      field_name: 'rule.name',
+    },
+    target_version: undefined,
+    merged_version: undefined,
+    diff_outcome: ThreeWayDiffOutcome.MissingBaseCanUpdate,
+    merge_outcome: ThreeWayMergeOutcome.Target,
+    has_update: true,
+    conflict: ThreeWayDiffConflict.SOLVABLE,
+  },
+};
+
+export const WithResolvedValue = Template.bind({});
+WithResolvedValue.args = {
+  fieldName: 'risk_score',
+  fieldThreeWayDiff: {
+    has_base_version: true,
+    base_version: 10,
+    current_version: 40,
+    target_version: 20,
+    merged_version: 40,
+    diff_outcome: ThreeWayDiffOutcome.CustomizedValueCanUpdate,
+    merge_outcome: ThreeWayMergeOutcome.Current,
+    has_update: true,
+    conflict: ThreeWayDiffConflict.NON_SOLVABLE,
+  },
+  resolvedValue: 35,
+};
+WithResolvedValue.argTypes = {
+  resolvedValue: {
+    control: 'number',
+  },
+};
+
+/* Optional field becomes defined - was undefined in base version, but was then defined by user in current version */
+export const OptionalFieldBecomesDefined = Template.bind({});
+OptionalFieldBecomesDefined.args = {
+  fieldName: 'timestamp_override',
+  fieldThreeWayDiff: {
+    has_base_version: true,
+    base_version: undefined,
+    current_version: {
+      field_name: 'event.ingested',
+      fallback_disabled: true,
+    },
+    target_version: undefined,
+    merged_version: {
+      field_name: 'event.ingested',
+      fallback_disabled: true,
+    },
+    diff_outcome: ThreeWayDiffOutcome.CustomizedValueNoUpdate,
+    merge_outcome: ThreeWayMergeOutcome.Current,
+    has_update: false,
+    conflict: ThreeWayDiffConflict.NONE,
+  },
+};
+
+export const SubfieldsWithLabels = Template.bind({});
+
+const subfieldsWithLabelsThreeWayDiff: ThreeWayDiff<RuleKqlQuery> = {
+  has_base_version: true,
+  base_version: {
+    type: KqlQueryType.inline_query,
+    query: 'event.agent_id_status: *',
+    language: 'kuery',
+    filters: [],
+  },
+  current_version: {
+    type: KqlQueryType.inline_query,
+    query: 'event.agent_id_status: *',
+    language: 'kuery',
+    filters: [],
+  },
+  target_version: {
+    type: KqlQueryType.saved_query,
+    saved_query_id: 'e355ef26-45f5-40f1-bbb7-5176ecf07d5c',
+  },
+  merged_version: {
+    type: KqlQueryType.saved_query,
+    saved_query_id: 'e355ef26-45f5-40f1-bbb7-5176ecf07d5c',
+  },
+  diff_outcome: ThreeWayDiffOutcome.StockValueCanUpdate,
+  merge_outcome: ThreeWayMergeOutcome.Target,
+  has_update: true,
+  conflict: ThreeWayDiffConflict.NONE,
```

Comment:
Actually my main idea was to just see / show how different field types (single / multi-line, numbers, arrays) are rendered. I didn't pay much attention to `conflict` since it doesn't directly affect this component.

Reply:
> Input data should be accurate.

I agree. And I think it is accurate in this case. 

`conflict` is `NONE`, because it's an `AAB` situation, so the target value is taken and there's no conflict. Isn't it?